chore: Make s3curl ft tests self-contained

DavidPineauScality opened the pull request:
* Add the required before/after steps for each describe block in the s3curl
tests. It was done this way so that we at least get a self-contained
granularity at the describe level rather than through the whole file

* Remove the data removal steps from circle.yml as it is not longer necessary

Fix #358 
Addresses part of #359 

Note: This work is a pre-requisite to an ongoing cleanup work aimed at improving dependency patcher suport in end to end tests.

alexandre-merle commented on tests/functional/s3curl/tests.js:
```diff
@@ -124,88 +122,155 @@ function provideRawOutput(args, cb) {
-            });
-    });
+function putObjects(filepath, objectPaths, cb) {
+    provideRawOutput([`--put=${filepath}`, '--', objectPaths[0], '-v'],
```
you could use aync here :)

alexandre-merle commented on tests/functional/s3curl/tests.js:
```diff
@@ -124,88 +122,155 @@ function provideRawOutput(args, cb) {
-            });
-    });
+function deleteObjects(objectPaths, cb) {
+    provideRawOutput(['--delete', '--', objectPaths[0], '-v'],
```
same here

alexandre-merle commented on tests/functional/s3curl/tests.js:
```diff
@@ -124,88 +122,155 @@ function provideRawOutput(args, cb) {
+describe('s3curl put delete buckets', () => {
+    describe('s3curl put buckets', () => {
+        after(done => {
+            provideRawOutput(['--delete', '--',
```
you can call deleteObjects here :)

alexandre-merle commented on tests/functional/s3curl/tests.js:
```diff
@@ -124,88 +122,155 @@ function provideRawOutput(args, cb) {
+        });
+
+        after(done => {
+            provideRawOutput(['--delete', '--',
```
same here, you can use the function :)

alexandre-merle commented on tests/functional/s3curl/tests.js:
```diff
@@ -124,88 +122,155 @@ function provideRawOutput(args, cb) {
+        });
+
+        it('should be able to delete a bucket', done => {
+            provideRawOutput(['--delete', '--',
```
same ^^

alexandre-merle commented on tests/functional/s3curl/tests.js:
```diff
@@ -124,88 +122,155 @@ function provideRawOutput(args, cb) {
-            `${endpoint}/${bucket}`, '-v'],
+describe('s3curl put and get bucket ACLs', () => {
+    after(done => {
+        provideRawOutput(['--delete', '--',
```
same :)

alexandre-merle commented on tests/functional/s3curl/tests.js:
```diff
@@ -278,6 +343,34 @@ describe('s3curl put and get bucket ACLs', () => {
+    });
+
+    after(done => {
+        provideRawOutput(['--delete', '--',
```
same ^^

alexandre-merle commented on tests/functional/s3curl/tests.js:
```diff
@@ -124,88 +122,155 @@ function provideRawOutput(args, cb) {
-                done();
-            });
-    });
+function putObjects(filepath, objectPaths, cb) {
```
this function can be use to delete objects and buckets, reducing a lot of tests :)

alexandre-merle commented:
so much better :D

michael-zapata commented on tests/functional/s3curl/tests.js:
```diff
@@ -301,7 +387,21 @@ describe('s3curl getService', () => {
+
+    after(done => {
+        deleteRemoteItems([
+            `${endpoint}/${bucket}/${prefix}${delimiter}${upload}1`,
```
The same goes for `` `${endpoint}/${bucket}/${prefix}${delimiter}${upload}` ``

LaurenSpiegel commented on tests/functional/s3curl/tests.js:
```diff
@@ -124,370 +127,570 @@ function provideRawOutput(args, cb) {
 });
 
 describe('s3curl head bucket', () => {
+    before('create file to put', done => {
```
create bucket

DavidPineauScality replied:
Ah, that reminds me that I didn't check the "before" and "after" messages. I guess I'll have to update this. Thanks

DavidPineauScality commented:
@LaurenSpiegel Didn't take a look yet at why the CI's failing, the only thing I see is that there is far too much log for CircleCI to show, I plan on doing that whenever I get some idle time from another task (hopefully tomorrow?)

michael-zapata commented:
Full error log:
```
 1) s3curl put delete buckets s3curl put buckets should put a valid bucket:

      Uncaught AssertionError: '409 CONFLICT' === '200 OK'
      + expected - actual

      -409 CONFLICT
      +200 OK
      
      at tests.js:186:28
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  2) s3curl put delete buckets s3curl put buckets "after all" hook:

      Uncaught AssertionError: '409 CONFLICT' === '204 NO CONTENT'
      + expected - actual

      -409 CONFLICT
      +204 NO CONTENT
      
      at tests.js:168:20
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  3) s3curl put delete buckets s3curl delete bucket "before all" hook:

      Uncaught AssertionError: '409 CONFLICT' === '200 OK'
      + expected - actual

      -409 CONFLICT
      +200 OK
      
      at tests.js:249:28
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  4) s3curl put delete buckets s3curl delete bucket "after all" hook:

      Uncaught AssertionError: '409 CONFLICT' === '204 NO CONTENT'
      + expected - actual

      -409 CONFLICT
      +204 NO CONTENT
      
      at tests.js:168:20
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  5) s3curl getService "before all" hook:

      Uncaught AssertionError: '409 CONFLICT' === '200 OK'
      + expected - actual

      -409 CONFLICT
      +200 OK
      
      at tests.js:378:24
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  6) s3curl getService "after all" hook:

      Uncaught AssertionError: '409 CONFLICT' === '204 NO CONTENT'
      + expected - actual

      -409 CONFLICT
      +204 NO CONTENT
      
      at tests.js:168:20
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  7) s3curl putObject "before all" hook: create file to put:

      Uncaught AssertionError: '409 CONFLICT' === '200 OK'
      + expected - actual

      -409 CONFLICT
      +200 OK
      
      at tests.js:422:24
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  8) s3curl getBucket "after all" hook:
     Uncaught AssertionError: [Error: Can't find line in http response code] === '204 NO CONTENT'
      at tests.js:168:20
      at ChildProcess.<anonymous> (tests.js:120:24)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  9) s3curl head bucket "before all" hook: create file to put:

      Uncaught AssertionError: '409 CONFLICT' === '200 OK'
      + expected - actual

      -409 CONFLICT
      +200 OK
      
      at tests.js:626:24
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  10) s3curl head bucket "after all" hook:

      Uncaught AssertionError: '409 CONFLICT' === '204 NO CONTENT'
      + expected - actual

      -409 CONFLICT
      +204 NO CONTENT
      
      at tests.js:168:20
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  11) s3curl getObject "before all" hook:

      Uncaught AssertionError: '409 CONFLICT' === '200 OK'
      + expected - actual

      -409 CONFLICT
      +200 OK
      
      at tests.js:661:28
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  12) s3curl getObject "after all" hook: delete created file and downloaded file:

      Uncaught AssertionError: '409 CONFLICT' === '204 NO CONTENT'
      + expected - actual

      -409 CONFLICT
      +204 NO CONTENT
      
      at tests.js:168:20
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  13) s3curl head object "before all" hook:

      Uncaught AssertionError: '409 CONFLICT' === '200 OK'
      + expected - actual

      -409 CONFLICT
      +200 OK
      
      at tests.js:711:28
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  14) s3curl head object "after all" hook: delete created file and downloaded file:

      Uncaught AssertionError: '409 CONFLICT' === '204 NO CONTENT'
      + expected - actual

      -409 CONFLICT
      +204 NO CONTENT
      
      at tests.js:168:20
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  15) s3curl object ACLs "before all" hook: create file to put:

      Uncaught AssertionError: '409 CONFLICT' === '200 OK'
      + expected - actual

      -409 CONFLICT
      +200 OK
      
      at tests.js:754:28
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  16) s3curl object ACLs "after all" hook: delete created file:

      Uncaught AssertionError: '409 CONFLICT' === '204 NO CONTENT'
      + expected - actual

      -409 CONFLICT
      +204 NO CONTENT
      
      at tests.js:168:20
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  17) s3curl multipart upload "before all" hook:

      Uncaught AssertionError: '409 CONFLICT' === '200 OK'
      + expected - actual

      -409 CONFLICT
      +200 OK
      
      at tests.js:873:24
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

  18) s3curl multipart upload "after all" hook:

      Uncaught AssertionError: '404 NOT FOUND' === '204 NO CONTENT'
      + expected - actual

      -404 NOT FOUND
      +204 NO CONTENT
      
      at tests.js:168:20
      at ChildProcess.<anonymous> (tests.js:126:16)
      at maybeClose (internal/child_process.js:821:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
```

DavidPineauScality commented:
@LaurenSpiegel Re-targetted rel/6.2.1 instead and rebased. Also removed the descriptive strings for "before" and "After" steps, as this is not necessary (mocha dumps enough info to find them without it)

LaurenSpiegel commented:
@ironman-machine check it

ironman-machine commented:
Hello @LaurenSpiegel

Starting end to end procedure using the following payload:
{"REPO_NAME": "S3", "SCALITY_S3_BRANCH": "fix/SelfContainedS3curl", "DEFAULT_BRANCH": "rel/6.2.1"}

Please follow http://ci.ironmann.io/gh/scality/Integration/6121 for CI status.
